src/test-run/index.js

```diff
@@ -3,19 +3,28 @@ import assert from 'assert';
 import { readSync as read } from 'read-file-relative';
 import Promise from 'pinkie';
 import Mustache from 'mustache';
+import uuid from 'uuid';
 import { Session } from 'testcafe-hammerhead';
 import TestRunDebugLog from './debug-log';
 import TestRunErrorFormattableAdapter from '../errors/test-run/formattable-adapter';
-import { TestDoneCommand, isTestDoneCommand, isCommandRejectableByPageError } from './commands';
+import {
+    TestDoneCommand,
+    TakeScreenshotOnFailCommand,
+    ExecutePendingBrowserManipulationCommand,
+    isTestDoneCommand,
+    isCommandRejectableByPageError,
+    isWindowManipulationCommand
+} from './commands';
 import CLIENT_MESSAGES from './client-messages';
 import STATE from './state';
 import COMMAND_TYPE from './commands/type';
 
 
 //Const
-const TEST_RUN_TEMPLATE               = read('../client/test-run/index.js.mustache');
-const TEST_DONE_CONFIRMATION_RESPONSE = 'test-done-confirmation';
-const MAX_RESPONSE_DELAY              = 2 * 60 * 1000;
+const TEST_RUN_TEMPLATE                   = read('../client/test-run/index.js.mustache');
+const TEST_DONE_CONFIRMATION_RESPONSE     = 'test-done-confirmation';
+const SCREENSHOT_CAN_NOT_BE_SAVED_MESSAGE = '[screenshot can not be saved by some reason]';
```

Comment:
I don't like `some reason` part. Maybe smthg like `was unable to save screenshot due to error`?